### PR TITLE
BC-6676 - Use v3 config

### PR DESCRIPTION
**Pull request by bischofmax**
# Short Description

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team
- [x] DEV: Every new component is implemented having accessibility in mind (e.g. aria-label, role property)

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -111,7 +111,8 @@ app.use(VueDOMPurifyHTML, {
 
 	initializeAxios(axios);
 
-	await envConfigModule.findEnvs();
+	const isDevelopment = process.env.NODE_ENV === "development";
+	await envConfigModule.loadConfiguration({ optional: isDevelopment });
```
[the remark this answers is not in the document]
Comment (uidp) [reply src/main.ts:119]: As I understand `optional === true` also means that the frontend will start without complaining when the core config is not available (not only the file config). Is that intended? Or should we differentiate which one actually has failed?

File: src/pages/administration/teacher-overview.unit.js [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
 	createTestingVuetify,
 } from "@@/tests/test-utils/setup";
-import BaseInput from "@/components/base/BaseInput/BaseInput.vue";
-import BaseLink from "@/components/base/BaseLink.vue";
-import BaseDialog from "@/components/base/BaseDialog/BaseDialog.vue";
-import BaseModal from "@/components/base/BaseModal.vue";
+import setupStores from "@@/tests/test-utils/setupStores";
 import { nextTick } from "vue";
+import { createStore } from "vuex";
+import mock$objects from "../../../tests/test-utils/pageStubs";
```
Comment (uidp) [inline src/pages/administration/teacher-overview.unit.js:19]: Maybe while we're at it: should be changed to `@@/tests/test-utils/pageStubs`

File: src/store/env-config.ts [hunk 3]
```diff
@@ -84,148 +151,173 @@ export default class EnvConfigModule extends VuexModule {
...
 	}
 
-	get getDefaultTimezone(): string {
-		return this.env.I18N__DEFAULT_TIMEZONE || "Europe/Berlin"; // TODO rely on server provided default
+	public get getDefaultTimezone(): Timezone {
+		return this.env.I18N__DEFAULT_TIMEZONE;
 	}
 
-	get getAdminToggleStudentLernstoreViewEnabled() {
+	get getAdminToggleStudentLernstoreViewEnabled(): boolean {
```
Comment (uidp) [inline src/store/env-config.ts:162]: Why was `public` only added to some of the getters/methods? I think when they're `public` by default we don't really have to write them but at least the code should be consistent.

File: src/store/env-config.ts [hunk 4]
```diff
@@ -84,148 +151,173 @@ export default class EnvConfigModule extends VuexModule {
...
-			Object.entries(requiredVars).forEach(([key]) => {
-				if (envs[key] == null) {
-					console.warn(`Missing configuration by server for key ${key}`);
-				}
-			});
-			this.setEnvs({ ...configsFromEnvironmentVars, ...envs });
+			const configCalls = [this.loadFileConfig(), this.loadCoreConfig()];
+
+			if (optional) {
+				await Promise.allSettled(configCalls);
```
Comment (uidp) [inline src/store/env-config.ts:304]: (See above comment regarding `optional === true`.)

File: src/components/organisms/administration/GeneralSettings.unit.js [hunk 5]
```diff
@@ -143,7 +144,7 @@ describe("GeneralSettings", () => {
 		schoolsModule.setSchool(school);
 
 		envConfigModule.setEnvs({
-			I18N__AVAILABLE_LANGUAGES: "de,en,es",
+			I18N__AVAILABLE_LANGUAGES: [Language.De, Language.En, Language.Es],
```
Comment (SevenWaysDP) [inline src/components/organisms/administration/GeneralSettings.unit.js:147]:
```suggestion
			I18N__AVAILABLE_LANGUAGES: [Language.De, Language.En, Language.Es].join(','),
```
???

File: src/components/organisms/administration/GeneralSettings.vue [hunk 6]
```diff
@@ -187,7 +187,7 @@ export default {
 			return schoolsModule.getLoading;
 		},
 		languages() {
-			return this.availableLanguages.split(",").map((lang) => {
+			return this.availableLanguages.map((lang) => {
```
Comment (SevenWaysDP) [inline src/components/organisms/administration/GeneralSettings.vue:190]: Zur obere Frage ob das einzige stelle ist wo wir als string bräuchten?

File: src/store/env-config.ts [hunk 7]
```diff
@@ -84,148 +151,173 @@ export default class EnvConfigModule extends VuexModule {
...
 
 	@Action
-	async findEnvs(): Promise<void> {
+	async loadConfiguration({ optional = false } = {}): Promise<void> {
 		try {
 			this.resetBusinessError();
 			this.setStatus("pending");
 
-			const envs = (await $axios.get("/v1/config/app/public")).data;
-			Object.entries(requiredVars).forEach(([key]) => {
```
[the remark this answers is not in the document]
Comment (CeEv) [reply src/store/env-config.ts:209]: Bewusst nicht. Wenn es wieder rein kommt wäre es nur schön das in einer extra Methode zu haben und evtl. nur für den Development Modus.

File: src/components/lern-store/LernstoreDetailView.vue [hunk 8]
```diff
@@ -282,7 +283,7 @@ export default {
 			return this.author && this.author !== DEFAULT_AUTHOR;
 		},
 		isBrandenburg() {
-			return process.env.SC_THEME === "brb";
+			return process.env.SC_THEME === SchulcloudTheme.Brb;
```
Comment (CeEv) [inline src/components/lern-store/LernstoreDetailView.vue:286]: Funktioniert die Methode überhaupt noch mit den Ausbau von Nuxt?

File: src/components/molecules/AdminTableLegend.vue [hunk 9]
```diff
@@ -61,11 +63,11 @@ export default {
 	computed: {
 		setInstituteTitle() {
 			switch (process.env.SC_THEME) {
```
Comment (CeEv) [inline src/components/molecules/AdminTableLegend.vue:65]: Guter Umbau, aber gleiche Fragen ...funktioniert das überhaupt? Woher kommt process.env im Browser des clients?

File: src/store/env-config.ts [hunk 10]
```diff
@@ -1,55 +1,96 @@
...
 	name: "envConfigModule",
 	namespaced: true,
 	stateFactory: true,
 })
 export default class EnvConfigModule extends VuexModule {
-	env: Envs = {
-		...requiredVars,
-		...configsFromEnvironmentVars,
-		ADMIN_TABLES_DISPLAY_CONSENT_COLUMN: true,
+	env: ConfigResponse = {
```
Comment (SevenWaysDP) [inline src/store/env-config.ts:31]: Kann das auch in extra Datei verschoben werden um dann im tests/test-utils/factory/envsFactory.ts zu verwenden, macht das Sinn?

File: src/main.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CeEv) [reply src/main.ts:122]: Is already resolved, but as hind it is possible with combination of Promise.all and Promise.settled if we like. But maybe not nessasary if you want to start the client without server it always fail.

File: tests/test-utils/factory/envsFactory.ts [hunk 11]
```diff
@@ -1,30 +1,68 @@
...
-	FALLBACK_DISABLED: false,
-	FEATURE_CONSENT_NECESSARY: true,
-	FEATURE_SCHOOL_SANIS_USER_MIGRATION_ENABLED: false,
-	FEATURE_LERNSTORE_ENABLED: false,
-	GHOST_BASE_URL: "",
-	I18N__AVAILABLE_LANGUAGES: "",
-	I18N__DEFAULT_LANGUAGE: "",
-	I18N__DEFAULT_TIMEZONE: "",
-	I18N__FALLBACK_LANGUAGE: "",
+export const envsFactory = Factory.define<ConfigResponse>(() => ({
```
Comment (CeEv) [inline tests/test-utils/factory/envsFactory.ts:9]: Es gibt einige Stellen wo man die factory hätte verwenden können.

**Review by CeEv — CHANGES_REQUESTED** [review]

**Review by CeEv — DISMISSED** [review]

**Review by CeEv — APPROVED** [review]
Bitte noch die 3 sonar Hinweise anschauen.

**Review by CeEv — APPROVED** [review]

**Pull request merged**